### PR TITLE
feat(extension-api): enhance `listImages()` method with optional options

### LINKED ISSUE #6256: feat(extension-api): enhancing listImage method with options
### Is your enhancement related to a problem? Please describe

Currently when listing images from an extension, we are using `listImages()` but we are not able to _filter_ what we want to list on.

### Describe the solution you'd like

Adding an `options?: {ListImageOptions}` argument.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Adding an optional argument to the `listImages` extension-api method. This PR is adding the option `provider` allowing to get list images for a specific provider.

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop/issues/6256

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by jeffmaury — APPROVED** [review]
LGTM

File: packages/extension-api/src/extension-api.d.ts [hunk 1]
```diff
@@ -2963,11 +2972,23 @@ declare module '@podman-desktop/api' {
...
-     * @return A promise resolving to an array of images information. This method returns a subset of the available information for images. To get the complete description of a specific image, you can use the {@link containerEngine.getImageInspect} method.
+     * @example
+     * // Example 1: List all container images when no specific provider is provided.
+     * const images = await listImages();
+     * console.log(images);
+     *
+     * @example
+     * // Example 2: List container images for a specific provider.
+     * const provider = provider.getContainerConnections().find(connection => connection.connection.status() === 'started');
+     * const images = await listImages({provider: provider });
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]:
```suggestion
     * const images = await listImages({provider});
```

File: packages/main/src/plugin/extension-loader.ts [hunk 2]
```diff
@@ -962,8 +962,8 @@ export class ExtensionLoader {
       ) {
         return containerProviderRegistry.buildImage(context, eventCollect, options);
       },
-      listImages(): Promise<containerDesktopAPI.ImageInfo[]> {
-        return containerProviderRegistry.listImages();
+      listImages(options?: containerDesktopAPI.ListImagesOptions): Promise<containerDesktopAPI.ImageInfo[]> {
+        return containerProviderRegistry.listImages(options);
```
Comment (benoitf) [inline packages/main/src/plugin/extension-loader.ts:966]: we should add a test in extension-loader.spec.ts

**Review by deboer-tim — APPROVED** [review]

File: packages/main/src/plugin/extension-loader.ts (hunk 2 above)
Comment (axel7083) [reply packages/main/src/plugin/extension-loader.ts:966]: Done ✅

**Pull request merged**